osb.go

```diff
@@ -0,0 +1,188 @@
+package tweed
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/pivotal-cf/brokerapi"
+)
+
+func (c *Core) OSB() http.Handler {
+	return brokerapi.New(c.OSBBroker(), lager.NewLogger("tweed"), brokerapi.BrokerCredentials{
+		Username: c.HTTPAuthUsername,
+		Password: c.HTTPAuthPassword,
+	})
+}
+
+func (c *Core) OSBBroker() brokerapi.ServiceBroker {
+	return broker{c: c}
+}
+
+type broker struct {
+	c *Core
+}
+
+func (b broker) Services(ctx context.Context) ([]brokerapi.Service, error) {
+	var services []brokerapi.Service
+	for _, s := range b.c.Config.Catalog.Services {
+		var servicePlans []brokerapi.ServicePlan
+		for _, p := range s.Plans {
+			sp := brokerapi.ServicePlan{
+				ID:          p.ID,
+				Name:        p.Name,
+				Description: p.Description,
+				Free:        &p.Free,
+				Bindable:    &p.Bindable,
+			}
+			servicePlans = append(servicePlans, sp)
+		}
+		dc := brokerapi.Service{
+			Name:                 s.Name,
+			ID:                   s.ID,
+			Description:          s.Description,
+			Tags:                 s.Tags,
+			Bindable:             s.Bindable,
+			InstancesRetrievable: s.InstancesRetrievable,
+			BindingsRetrievable:  s.BindingsRetrievable,
+			PlanUpdatable:        s.PlanUpdateable,
+			Plans:                servicePlans,
+		}
+		services = append(services, dc)
+	}
+	return services, nil
+}
+
+func (b broker) Provision(ctx context.Context, instanceID string, details brokerapi.ProvisionDetails, asyncAllowed bool) (brokerapi.ProvisionedServiceSpec, error) {
+	// FIXME: Uncomment after prefixing is done
+	// if err := ValidInstanceID(instanceID); err != nil {
+	// 	return brokerapi.ProvisionedServiceSpec{}, err
+	// }
```

Comment:
Let's fix the fixme here, and make ValidInstanceID happy with leading numeric digits (for CF's sake)